### ISSUE TITLE
gatsby-rehype-ghost-links doesn't seem to update GhostPage items

### ISSUE THREAD

**Issue by shawnpyle** [styxlab/gatsby-theme-try-ghost#49]
Thanks for your work with all things Ghost. I'm having a hard time getting absolute links to be converted to relative using gatsby-rehype-ghost-links.

This is what I've done:
1. Cloned a copy of https://github.com/styxlab/gatsby-starter-try-ghost 
1. Update my `.ghost.json` with my ghost server
1. `gatsby develop`

I'm seeing that the post content gets relative links but not the pages. I've tried 
1. debugging the plugin and spitting out URLs that it's processing during the build, but it appears to be working correctly in that page content is getting run through the gatsby-rehype-ghost-links/gatsby-node.js functionality. 
1. gatsby repl but the nodes don't have updated links which I'm assuming occurs after the 
1. Plinking around with the  graphql

I thought Netlify might build it differently so I sent it there but I'm seeing the same thing, relative links in the posts but not in the pages. 

Any additional ideas on how to debug why pages aren't getting processed properly? As you can see I'm out of my league. Thank you for your pity. 😞

**Comment by styxlab** [comment]
If you use the standard filter in `gatsby-config.js`:
```
        {
            resolve: `gatsby-transformer-rehype`,
            options: {
                filter: node => (
                    node.internal.type === `GhostPost` ||
                    node.internal.type === `GhostPage`
                ) && node.slug !== `data-schema`,
                plugins: [
                    {
                        resolve: `gatsby-rehype-ghost-links`,
                    },
                    {
                        resolve: `gatsby-rehype-prismjs`,
                    },
                ],
            },
        },
```
`GhostPage` is treated exactly the same way as `GhostPost`.

Could you please publish your content API keys here and tell me exactly the links that are not working, so I can debug further?

**Comment by shawnpyle** [comment]
Sorry, I should have done that from the get go. 

```
$ nvm version
v12.16.2
```
```
$ gatsby -v
Gatsby CLI version: 2.11.11
Gatsby version: 2.21.1
```
```
$ cat .ghost.json
{
  "development": {
    "apiUrl": "https://ghost.coachwithdesign.com",
    "contentApiKey": "cce1f0e60ec7d0fb55effe5f41"
  },
  "production": {
    "apiUrl": "https://ghost.coachwithdesign.com",
    "contentApiKey": "cce1f0e60ec7d0fb55effe5f41"
  }
}
```
```
$ cat gatsby-config.js
let siteConfig
let ghostConfig

try {
    siteConfig = require(`./siteConfig`)
} catch (e) {
    siteConfig = null
}

try {
    ghostConfig = require(`./.ghost`)
} catch (e) {
    ghostConfig = {
        development: {
            apiUrl: process.env.GHOST_API_URL,
            contentApiKey: process.env.GHOST_CONTENT_API_KEY,
        },
        production: {
            apiUrl: process.env.GHOST_API_URL,
            contentApiKey: process.env.GHOST_CONTENT_API_KEY,
        },
    }
} finally {
    const { apiUrl, contentApiKey } = process.env.NODE_ENV === `development` ? ghostConfig.development : ghostConfig.production

    if (!apiUrl || !contentApiKey || contentApiKey.match(/<key>/)) {
        ghostConfig = null //allow default config to take over
    }
}

module.exports = {
    plugins: [
        {
            resolve: `gatsby-theme-try-ghost`,
            options: {
                ghostConfig: ghostConfig,
                siteConfig: siteConfig,
            },
        },
        {
            resolve: `gatsby-theme-ghost-dark-mode`,
            options: {
                // Set to true if you want your theme to default to dark mode (default: false)
                // Note that this setting has an effect only, if
                //    1. The user has not changed the dark mode
                //    2. Dark mode is not reported from OS
                defaultModeDark: false,
                // If you want the defaultModeDark setting to take precedence
                // over the mode reported from OS, set this to true (default: false)
                overrideOS: false,
            },
        },
        {
            resolve: `gatsby-transformer-rehype`,
            options: {
                filter: node => (
                    node.internal.type === `GhostPost` ||
                    node.internal.type === `GhostPage`
                ) && node.slug !== `data-schema`,
                plugins: [
                    {
                        resolve: `gatsby-rehype-ghost-links`,
                    },
                    {
                        resolve: `gatsby-rehype-prismjs`,
                    },
                ],
            },
        },
    ],
}
```
```
$ cat siteConfig.js
module.exports = {
    siteUrl: `https://coachwithdesign.com`, // Site domain. Do not include a trailing slash!

    postsPerPage: 12, // Number of posts shown on paginated pages (changes this requires sometimes to delete the cache)

    siteTitleMeta: `Coach with Design`, // This allows an alternative site title for meta data for pages.
    siteDescriptionMeta: `A blog about understanding who we are and finding purpose in our design.`, // This allows an alternative site description for meta data for pages.

    shareImageWidth: 1000, // Change to the width of your default share image
    shareImageHeight: 523, // Change to the height of your default share image

    shortTitle: `CwD`, // Used for App manifest e.g. Mobile Home Screen
    siteIcon: `favicon.png`, // Logo in /static dir used for SEO, RSS, and App manifest
    backgroundColor: `#e9e9e9`, // Used for Offline Manifest
    themeColor: `#15171A`, // Used for Offline Manifest
}
```

**Comment by styxlab** [comment]
Thanks for providing detailed information. I have reproduced the issue internally and confirm this bug.

I will provide a fix later today. Thanks for catching this!

**Comment by styxlab** [comment]
Fixed in versions `>= 1.9.1`.

**Comment by shawnpyle** [comment]
Perfect! Confirm it works as expected. Thank you very much.
